source/agora/test/SlashingMisbehavingValidator.d

```diff
@@ -52,10 +53,10 @@ private class MissingPreImageEM : EnrollmentManager
     public override bool getNextPreimage (out PreImageInfo preimage,
         Height height) @safe
     {
-        if (*this.reveal_preimage == false)
+        if (!atomicLoad(*this.reveal_preimage))
```

Comment:
What are advantages of using atomic operations here, and in L229?

Reply:
Avoiding bugs ;)

Jokes aside, it doesn't matter too much here, because it's  a simple boolean that is flipped, but you should never access shared data without proper synchronization. The basis of which is atomic operations. So a more serious answer would be "correctness".